llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -992,6 +993,8 @@ void AMDGPUPassConfig::addIRPasses() {
   if (isPassEnabled(EnableImageIntrinsicOptimizer))
     addPass(createAMDGPUImageIntrinsicOptimizerPass(&TM));
 
+  addPass(createExpandVariadicsPass(ExpandVariadicsMode::Lowering));
```

Comment:
Please add a kill switch `cl::opt` for the pass in the AMDGPUTargetMachine before landing, in case it has unforeseen issues. It'd be easier to default-disable while fixing, rather than revert a big patch :)

Reply:
I'm also paranoid enough to want that, it's '--expand-variadics-override=disable' from the command line or changing the ::Lowering in that line to ::Disable. Commenting out the addPass also kills it, but would cause the lit pipeline tests to fail. I'll put a comment above that pass saying this.